modules/loader-utils/src/lib/iterator-utils/async-iteration.js

```diff
@@ -38,14 +38,23 @@ export async function forEach(iterator, visitor) {
  * This function can e.g. be used to enable atomic parsers to work on (async) iterator inputs
  */
 export async function concatenateChunksAsync(asyncIterator) {
-  let arrayBuffer = new ArrayBuffer(0);
-  let string = '';
+  const arrayBuffers = [new ArrayBuffer(0)];
+  const strings = [''];
```

Comment:
why are these not just empty to start?

Reply:
Wanted to handle the case where the input iterator is empty and has no batches.

Reply:
Could just create the array buffer at the end of the function then. This always concatenates an additional empty buffer.

Reply:
Yes you are right of course, updated. 

(I was so focused on optimizing this for 100 MB concatenations that I didn't give much attention to the perf in the simple cases.)